hosted-cluster-config-operator/operator/config.go

```diff
@@ -198,6 +204,32 @@ func (c *HostedClusterConfigOperatorConfig) TargetKubeInformersForNamespace(name
 	return informer
 }
 
+func (c *HostedClusterConfigOperatorConfig) ControlPlaneCluster() cluster.Cluster {
+	if c.cpCluster == nil {
+		var err error
+		c.cpCluster, err = cluster.New(c.Config(), withScheme(api.Scheme), withNamespace(c.namespace))
+		if err != nil {
+			c.Fatal(err, "cannot get control plane cluster client")
+		}
+		c.Manager().Add(manager.RunnableFunc(func(ctx context.Context) error {
```

Comment:
You need to directly add the `cpCluster` rather than wrapping it like that, so the manager can ensure its cache is synced before other runnables are started